src/Symfony/Component/OptionsResolver/OptionConfigurator.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver;
+
+use Symfony\Component\OptionsResolver\Exception\AccessException;
+
+final class OptionConfigurator
+{
+    private $name;
+
+    private $resolver;
+
+    public function __construct(string $name, OptionsResolver $resolver)
+    {
+        $this->name = $name;
+        $this->resolver = $resolver;
+        $this->resolver->setDefined($name);
+    }
+
+    /**
+     * Adds allowed values for this option.
+     *
+     * @param mixed $allowedValues One or more acceptable values/closures
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addAllowedValues($allowedValues): self
+    {
+        $this->resolver->addAllowedValues($this->name, $allowedValues);
+
+        return $this;
+    }
+
+    /**
+     * Sets allowed types for this option.
+     *
+     * @param string|string[] $allowedTypes One or more accepted types
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addAllowedTypes($allowedTypes): self
```

Comment:
All these methods that accept a single or several values  should use variadic arguments IMO. We didn't do that in the OptionsResolver because it was not available back then.